kythe/go/extractors/gcp/config/converter.go

```diff
@@ -80,30 +80,20 @@ func KytheToBuild(conf *rpb.Config) (*cloudbuild.Build, error) {
 				Paths:    []string{path.Join(outputDirectory, outputFilePattern)},
 			},
 		},
+		// TODO(danielmoy): this should probably also be a stepper, or at least
+		// if there is refactoring work done as described below to make steps
+		// more granular, this will have to hook into that logic.
 		Steps: commonSteps(),
 	}
 
 	hints := conf.Extractions[0]
 
-	// TODO(danielmoy): when we support bazel, we probably want to pull out the
-	// switch cases here into something a bit easier to read.
-	// I imagine something like a proper interface that each build system can
-	// override, which would provide things like artifactPaths(), steps().
-	switch hints.BuildSystem {
-	case rpb.BuildSystem_MAVEN:
-		build.Steps = append(build.Steps,
-			javaArtifactsStep(),
-			mavenStep(hints))
-		build.Artifacts.Objects.Paths = append([]string{path.Join(outputDirectory, "javac-extractor.err")}, build.Artifacts.Objects.Paths...)
-	case rpb.BuildSystem_GRADLE:
-		build.Steps = append(build.Steps,
-			javaArtifactsStep(),
-			gradleStep(hints))
-		build.Artifacts.Objects.Paths = append([]string{path.Join(outputDirectory, "javac-extractor.err")}, build.Artifacts.Objects.Paths...)
-	default:
-		return build, fmt.Errorf("unsupported build system %s", hints.BuildSystem)
+	st, err := stepper(hints.BuildSystem)
+	if err != nil {
+		return build, err
```

Comment:
Why not return nil, err? Is there something the user is supposed to do with build in the case of an error?